crates/node/p2p/src/net/builder.rs

```diff
@@ -36,36 +32,51 @@ pub struct NetworkBuilder {
 
 impl From<Config> for NetworkBuilder {
     fn from(config: Config) -> Self {
-        Self::new()
-            .with_discovery_config(config.discovery_config)
-            .with_discovery_randomize(config.discovery_randomize)
-            .with_bootstore(config.bootstore)
-            .with_bootnodes(config.bootnodes)
-            .with_discovery_interval(config.discovery_interval)
-            .with_discovery_address(config.discovery_address)
-            .with_gossip_address(config.gossip_address)
-            .with_unsafe_block_signer(config.unsafe_block_signer)
-            .with_gossip_config(config.gossip_config)
-            .with_peer_scoring(config.scoring)
-            .with_peer_monitoring(config.monitor_peers)
-            .with_block_time(config.block_time)
-            .with_keypair(config.keypair)
-            .with_topic_scoring(config.topic_scoring)
-            .with_gater_config(config.gater_config)
-            .with_local_signer(config.local_signer)
+        Self::new(
+            config.rollup_config,
+            config.unsafe_block_signer,
+            config.gossip_address,
+            config.keypair,
+            config.discovery_address,
+            config.discovery_config,
+        )
+        .with_discovery_randomize(config.discovery_randomize)
+        .with_bootstore(config.bootstore)
+        .with_bootnodes(config.bootnodes)
+        .with_discovery_interval(config.discovery_interval)
+        .with_gossip_config(config.gossip_config)
+        .with_peer_scoring(config.scoring)
+        .with_peer_monitoring(config.monitor_peers)
+        .with_topic_scoring(config.topic_scoring)
+        .with_gater_config(config.gater_config)
+        .with_local_signer(config.local_signer)
```

Comment:
It's an aesthetic thing but I find this to be a bit of an unappealing user interface personally. Not a big deal though.

Reply:
Totally correct! I'm fine merging this and going this route